Stringfix & get_utf8 method cleanup

walkerjoe opened the pull request:
- Ensuring mock data is binary type before write to fp
- Duplicate `get_utf8_value` methods found in boto/utils.py

walkerjoe committed:
Ensure correct type for writing mocked data

walkerjoe committed:
removing duplicate `get_utf8_value` method

catleeball commented:
LGTM! Sorry about messing up with git a doubling the `get_utf8_vaule` function, thanks for catching that. :)

houglum reviewed (approved):
LGTM

houglum commented:
@mfschwartz Could you merge this when you get the chance? Thanks!

walkerjoe committed:
Added idna==2.7 to py26 and py33 req docs

idna 2.8 requires 2.7 or 3.4+ in order to run. idna version 2.7,
however, does support the older versions of python so the tests can
be performed.

walkerjoe committed:
Adding PyYAML==3.1 for python 2.6 compatibility
